pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java

```diff
@@ -253,77 +131,155 @@ public void generate(final Protobuf3Parser.MessageDefContext msgDef,
+
+		// generate builder
+		bodyContent += generateBuilder(msgDef, fields, lookupHelper);
+		bodyContent += "\n";
+
+		// oneof enums
+		bodyContent += String.join("\n    ", oneofEnums);
+
+		// === Build file
+		try (final FileWriter javaWriter = new FileWriter(javaFile)) {
+			javaWriter.write(
+					generateClass(modelPackage, imports, javaDocComment, deprecated, javaRecordName, fields, bodyContent)
+			);
+		}
+	}
+
+	/**
+	 * Generating method that assembles all the previously generated pieces together
+	 * @param modelPackage the model package to use for the code generation
+	 * @param imports the imports to use for the code generation
+	 * @param javaDocComment the java doc comment to use for the code generation
+	 * @param deprecated the deprecated annotation to add
+	 * @param javaRecordName the name of the class
+	 * @param fields the fields to use for the code generation
+	 * @param bodyContent the body content to use for the code generation
+	 * @return the generated code
+	 */
+	@NotNull
+	private static String generateClass(final String modelPackage,
+										final Set<String> imports,
+										final String javaDocComment,
+										final String deprecated,
+										final String javaRecordName,
+										final List<Field> fields,
+										final String bodyContent) {
+		return """
+				package $package;
+				$imports
+				import com.hedera.pbj.runtime.Codec;
+				import java.util.function.Consumer;
+				import java.lang.Comparable;
+				import edu.umd.cs.findbugs.annotations.Nullable;
+				import static java.util.Objects.requireNonNull;
+									
+				$javaDocComment$deprecated
+				public record $javaRecordName(
+				$fields) implements Comparable<$javaRecordName> {
+				$bodyContent}
+				"""
+				.replace("$package", modelPackage)
+				.replace("$imports", imports.isEmpty() ? "" : imports.stream().collect(Collectors.joining(".*;\nimport ", "\nimport ", ".*;\n")))
+				.replace("$javaDocComment", javaDocComment)
+				.replace("$deprecated", deprecated)
+				.replace("$javaRecordName", javaRecordName)
+				.replace("$fields", fields.stream().map(field ->
+						(field.type() == FieldType.MESSAGE ? "@Nullable " : "")
+								+ field.javaFieldType() + " " + field.nameCamelFirstLower()
+				).collect(Collectors.joining(",\n")).indent(DEFAULT_INDENT))
+				.replace("$bodyContent", bodyContent);
+	}
+
+
+	/**
+	 * Filter the fields to only include those that are comparable
+	 * @param msgDef The message definition
+	 * @param lookupHelper The lookup helper
+	 * @param fields The fields to filter
+	 * @return the filtered fields
+	 */
+	@NonNull
+	private static List<Field> filterComparableFields(final MessageDefContext msgDef,
+													  final ContextualLookupHelper lookupHelper,
+													  final List<Field> fields) {
+		final Set<String> comparableFields = lookupHelper.getComparableFields(msgDef);
+		if(comparableFields.isEmpty()) {
+			// consider all fields in compareTo by defalut
```

Comment:
This seems a bit extreme, especially if the current implementation doesn't support comparisons of certain field types as discussed above. I'd suggest not to implement Comparable for models that don't have the optional annotation comment added explicitly. This would simplify the implementation as well as reduce the footprint of the generated Java classes (both source, and more importantly, the binaries) for the model codecs.